### ISSUE TITLE
Issue of crown depth

### ISSUE THREAD

**Issue by JunyanDing** [NGEET/fates#674]
In fire module the parameter fates_fire_crown_depth_frac is used to calculate the crown depth
 https://github.com/NGEET/fates/blob/master/fire/SFMainMod.F90#L885-L898

But in allometry module the crown depth is calculated as the smaller value of tree height and 0.1m:
https://github.com/NGEET/fates/blob/master/biogeochem/FatesAllometryMod.F90#L1910 

Should we calculate the crown depth the same way as in fire module? Also, in fire module this parameter is named as EDPftvarcon_inst%crown, which seems informative.

**Comment by ckoven** [comment]
Looking at the code, the subroutine `CrownDepth` is only used by the hydraulics code, does it have the same physical/ecological meaning as in the fire code?  If so, I agree that we ought to unify those calculations.

**Comment by JunyanDing** [comment]
It affects the length of the stem, hence the total hydraulic conductance of the tree, thus the leaf water potential.

**Comment by rosiealice** [comment]
That seems to me like it's -not- the same definition as the fire module,which is trying to calculate the vertical height over which there are leaves (as opposed to stem). This might be an issue of definitions and the hydraulics one sounds more like just 'tree height' to me. Although, admittedly, most water doesn't have to actually reaxh the very top of the tree so you could imagine unifying the concepts a little more....

**Comment by ckoven** [comment]
ok, @rosiealice, I see your point. But I still think there may be an error on https://github.com/NGEET/fates/blob/master/biogeochem/FatesAllometryMod.F90#L1910:
shouldn't it be `crown_depth = max(height,0.1_r8)`?

**Comment by rosiealice** [comment]
Ah yes, indeed...

**Comment by rosiealice** [comment]
That might cause quite a bit of answer-changing...

**Comment by ckoven** [comment]
yeah.  @xuchongang, @bchristo, @rgknox: thoughts on this?

**Comment by JunyanDing** [comment]
@ckoven and @rosiealice , I disagree with both of you 😉 
That line of code is on allometry module, which means FATES treats crown depth conceptually differently for fire and the rest of other processes though ut is only used in hydro currently. 
Stem height is calculated as tree height - crown depth. If set to  max(height, 0.1), the height of stem will either be zero if tree is shorter than 0.1m or negative if the tree is taller than 0.1m

**Comment by JunyanDing** [comment]
For hydro , I will set the height of stem to be the middle of the crown. I think this is the appropriate height(or the length of flow path) for all the leaves in average

**Comment by jkshuman** [comment]
I agree that these are fundamentally different concepts. @rosiealice is right that fire is explicitly looking at crown depth as the actual depth of foliage or distance between top of tree and clear branch bole height which we are calling the base of the crown. This is made more clear in the implementation of crown fire, but perhaps I will clean this up in the base code before finishing the crown fire implementation.

```real(r8) ::  crown_depth          ! depth of crown (m)
     real(r8) ::  height_cbb           ! clear branch bole height or crown base height (m)

     crown_depth   = currentCohort%hite*EDPftvarcon_inst%crown(currentCohort%pft) 
     height_cbb     = currentCohort%hite - crown_depth```

https://github.com/jkshuman/fates/blob/339aca2172839cc3aed5427db8af6982e8170d9a/fire/SFMainMod.F90#L934-L935

It sounds like the Hydro use of crown depth is different, and perhaps this name should be updated so that it is more clear. From @JunyanDing it sounds like this is a stem height to middle of the crown?

**Comment by jkshuman** [comment]
Also, the way this is currently written it seems that this value will give you a value that is just less than the total plant height? (this is in line with @JunyanDing assessment, and her suggested change to the middle of the crown.)
with crown depth = min(height, 0.1)

 ` real(r8) :: z_stem        ! the height of the plants stem below crown [m]`

https://github.com/NGEET/fates/blob/68bc434451cbc4981745c715497feb7fe05d042a/biogeophys/FatesPlantHydraulicsMod.F90#L674

so the use of crown_depth for Hydro taken from allometry.mod is not the same as in fire, but the comments on z_stem suggest it should be using crown_depth to get base crown height (same as in fire). 

The parameterization of EDPftvarcon_inst%crown(currentCohort%pft) is at the pft level, and is set in the param file. 

https://github.com/NGEET/fates/blob/68bc434451cbc4981745c715497feb7fe05d042a/parameter_files/fates_params_default.cdl#L173-L175

Is this in line with how it is intended to be used in Hydro? @xuchongang @JunyanDing @bchristo @rgknox

**Comment by JunyanDing** [comment]
Your are right @jkshuman  The crown depth is used in the same way in Hydro as it used in fire module. 
I think to change code of line 674 in FatesPlantHydraulicsMod.F90
z_stem = plant_height - crown_depth
to  
z_stem  = currentCohort%hite - 0.5* (currentCohort%hite*EDPftvarcon_inst%crown(currentCohort%pft))

**Comment by jkshuman** [comment]
ok. If the change that @JunyanDing is proposing is implemented, I suggest we also change the name of the parameter "fates_fire_crown_depth_frac" in the parameter file to remove the "fire" tag, and just make it generic to "fates_crown_depth_frac"

**Comment by JunyanDing** [comment]
I agree with @jkshuman. And we can just remove the subroutine
 subroutine CrownDepth(height,crown_depth) in FatesAllometryMod.F90
https://github.com/NGEET/fates/blob/master/biogeochem/FatesAllometryMod.F90#L1893

As in Hydro module, if it is more appropriate to set the stem height to close to the top of the tree, we can just set  z_stem = currentCohort%hite - 0.2(or 0.1)* (currentCohort%hite*EDPftvarcon_inst%crown(currentCohort%pft))

**Comment by rgknox** [comment]
The hydraulics code needs the height of the center of volume for each compartment. It is used in calculating the flux between compartments, by 1) setting the mean distance* of conductance between compartments, it also 2) sets the elevation potential that has to be overcome by suction.  If the leaf area density were uniformly distributed in the vertical coordinate, it would seem reasonable to set the center of volume at halfway down the depth of the crown.   But that seems unlikely, no?  

After reviewing the code, as of sci.1.59.1_api.24.1.0, @ckoven @glemieux and I saw that [fire](https://github.com/NGEET/fates/blob/sci.1.59.1_api.24.1.0/fire/SFMainMod.F90#L929) is using the [CrownDepth](https://github.com/NGEET/fates/blob/sci.1.59.1_api.24.1.0/biogeochem/FatesAllometryMod.F90#L1989) function and file-based [parameter](https://github.com/NGEET/fates/blob/sci.1.59.1_api.24.1.0/parameter_files/fates_params_default.cdl#L876), yet hydro is still using 0.1 ([here](https://github.com/NGEET/fates/blob/sci.1.59.1_api.24.1.0/biogeophys/FatesPlantHydraulicsMod.F90#L777-L778) and [here](https://github.com/NGEET/fates/blob/sci.1.59.1_api.24.1.0/biogeophys/FatesPlantHydraulicsMod.F90#L1006-L1007)).

*Also, the distance that must be overcome for conductance, does not take into account the mean horizontal distance travelled from the stem to the leaf.  It seems like it should take this into account.

@JunyanDing @xuchongang @bchristo @pnlfang

**Comment by JunyanDing** [comment]
I think if we leave the crowndepth in hydro to be 0.1, it might just take into account the horizontal distance from stem to leaf (length of branch).

**Comment by bchristo** [comment]
Junyan is correct. The assumption (pipe model) is that the path length from
stem base to branch tip is the same for all branches. The model
approximates the tree as a beam -- while this gets the hydraulic path
length "correct" (in line with assumption), technically it overestimates
the gravity effect since the vertical height of most branch tips will be
less than tree height.

**Comment by rgknox** [comment]
So are you proposing the vertical coordinate for the center of crown be:

height - 0.1     or     height - crown_depth * 0.1

I'm thinking of hemlocks, firs and pines and how many of them distribute their crowns over much of their height.  It seems like option one would be asking them to overcome more resistance and potential than necessary.  I realize though, that we are trying to keep things simple, and not "branch out" (pun intended), into sophisticated crown models that require more information on crown geometry.

**Comment by bchristo** [comment]
Hydro from the beginning has always approximated the canopy as a flat disc
(neglible crown depth) at the top of a vertical beam, so in that sense
option 1 aligns most closely to that.
That said, some of the large sensitivity to the xylem conduit taper
exponent we are observing (Chonggang, Charlie, Zachary) may be related to
precisely the problem you highlight. For the sake of this thread, option 1
is best, but may be prudent to open a separate issue to look at this in a
broader context of some of the simulated dynamics.

**Comment by rosiealice** [comment]
Is it possible those trees make branches so low down because of hydraulic limitation?? 

Or probably more likely due to mechanical stability. Still interesting point. I agree with Brad that this crown geometry thing might be its own issue...

**Comment by rgknox** [comment]
I think the way around this is to change our language for how this "depth" impacts hydraulics. Instead of defining this depth as something about biomass, its really used to assign the depth to the center of hydraulic potential, or rather where we prognose pressure in the leaves at a point. 
Proposal, add a new parameter and change an old parameter:

Name change: fates_allom_crown_depth_frac -> fates_fire_crown_depth_frac  

New parameter:  fates_hydro_leafpressure_depth

**Comment by JunyanDing** [comment]
@rgknox I like this idea. That makes sense.
